### PR TITLE
A0-2433: update substrate to 0.9.39

### PR DESCRIPTION
**Pull request by kostekIV**
update to substrate 0.9.39.

The liminal build in particular the baby-liminal-extension was broken. I rm it as a dependency from runtime, plus commented out ci step checking it. CC @pmikolajczyk41.

### CODE REVIEW

**Comment by pmikolajczyk41** [comment]
I suggest rebasing onto the latest commit on main, hopefully you'll find it useful :pray:

**Comment by kostekIV** [comment]
> I suggest rebasing onto the latest commit on main, hopefully you'll find it useful pray

well, its required so I have to :|

File: bin/runtime/Cargo.toml [hunk 1]
```diff
@@ -30,7 +30,7 @@ pallet-identity = { workspace = true }
 pallet-multisig = { workspace = true }
 pallet-nomination-pools = { workspace = true }
 pallet-nomination-pools-runtime-api = { workspace = true }
-pallet-randomness-collective-flip = { workspace = true }
+pallet-insecure-randomness-collective-flip = { workspace = true }
```
Comment (pmikolajczyk41) [inline bin/runtime/Cargo.toml:33]: here: https://github.com/paritytech/substrate/tree/master/frame/insecure-randomness-collective-flip they are screaming that: **DO NOT USE IN PRODUCTION** - should we be worried? or it has always been like this

**Review by pmikolajczyk41 — APPROVED** [review]
lgtm, gg

File: bin/runtime/Cargo.toml (hunk 1 above)
Comment (kostekIV) [reply bin/runtime/Cargo.toml:33]: it has always been like this. afaik it is only used in the pallet_contracts and source of randomness

Comment (mike1729) [reply bin/runtime/Cargo.toml:33]: Is it still? I believe there was an effort to remove it after last audit.

Comment (kostekIV) [reply bin/runtime/Cargo.toml:33]: yep, this is deprecated so we dont have to worry about it being insecure. I can remove the randomness pallet from the runtime but lets do it in separate pr
image

Comment (pmikolajczyk41) [reply bin/runtime/Cargo.toml:33]: sounds good for me, resolving

File: finality-aleph/src/party/manager/mod.rs [hunk 2]
```diff
@@ -333,6 +330,11 @@ where
         };
 
         let last_block_of_previous_session = session_boundaries.first_block().saturating_sub(1);
+        let last_block_of_previous_session_hash = self
+            .client
+            .block_hash(last_block_of_previous_session)
+            .unwrap()
+            .unwrap();
```
Comment (mike1729) [inline finality-aleph/src/party/manager/mod.rs:337]: I'm afraid that two unwraps don't work as two minuses :(

**Review by mike1729 — CHANGES_REQUESTED** [review]
Unwrap two unwraps.

File: finality-aleph/src/party/manager/mod.rs (hunk 2 above)
Comment (kostekIV) [reply finality-aleph/src/party/manager/mod.rs]: Changed to expect :)

**Review by mike1729 — APPROVED** [review]

**Pull request merged**